src/robot/model/body.py

```diff
@@ -43,7 +54,7 @@ class BodyItem(ModelObject):
     __slots__ = ['parent']
 
     @property
-    def id(self):
+    def id(self) -> 'str|None':
```

Comment:
Returning an empty string feels like the most sensible approach.  But there is a comment that makes me apprehensive about changing it. 

```py
# This algorithm must match the id creation algorithm in the JavaScript side
# or linking to warnings and errors won't work.
```

Reply:
On Javascript side the structure is flattened so that there are no `If` or `Try` roots.

Because `id` should never be accessed from `If/Try` roots, I believe it's best to change them to raise a TypeError in that case.

Reply:
The TypeChecker mainly complains about the following clause 
```
    @property
    def id(self) -> 'str':
        if not self:
            return None
```

I'll update this to return an empty string

Reply:
Changing this to return a string caused test failure, So I've reverted.